### PR TITLE
🐛 561 - Modify update user logic

### PR DESCRIPTION
**Pull request by anncatton**

### CODE REVIEW

**Review by blabadi — APPROVED** [review]

File: src/main/java/bio/overture/ego/model/dto/UpdateUserRequest.java [hunk 1]
```diff
@@ -31,6 +32,8 @@
   private UserType type;
   private StatusType status;
   private LanguageType preferredLanguage;
-  @NonNull private ProviderType providerType;
-  @NonNull private String providerSubjectId;
+  private String firstName;
+  private String lastName;
+  @NotNull private ProviderType providerType;
```
Comment (andricDu) [inline src/main/java/bio/overture/ego/model/dto/UpdateUserRequest.java:37]: Do we want to be able to change provider information via REST API?

**Review by andricDu — CHANGES_REQUESTED** [review]

File: src/main/java/bio/overture/ego/model/dto/UpdateUserRequest.java (hunk 1 above)
Comment (anncatton) [reply src/main/java/bio/overture/ego/model/dto/UpdateUserRequest.java:37]: I included the provider info here to add an extra validation on the user's provider sub id & type when updating:
https://github.com/overture-stack/ego/blob/8bd699b784bb442458239f150cb073450e3f1521/src/main/java/bio/overture/ego/service/UserService.java#L511
It was my intention to not allow the provider info to be changed, the request would error out if the incoming user provider info didn't match our stored user. But maybe I missed something?

File: src/main/java/bio/overture/ego/controller/UserController.java [hunk 2]
```diff
@@ -146,7 +146,7 @@ public UserController(
   }
 
   @AdminScoped
-  @RequestMapping(method = RequestMethod.PUT, value = "/{id}")
+  @RequestMapping(method = RequestMethod.PATCH, value = "/{id}")
```
Comment (andricDu) [inline src/main/java/bio/overture/ego/controller/UserController.java:149]: :fireworks:

**Review by andricDu — APPROVED** [review]
Thanks!

**Pull request merged**